### ISSUE TITLE
Do not enforce line length limit for long URL in docstrings

### ISSUE THREAD

**Issue by jfcherng** [charliermarsh/ruff#3051]
Similar to https://github.com/charliermarsh/ruff/pull/920 but in docstring

```py
class Foo:
    """
    @see https://looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong.url.com
    """
```

**Comment by charliermarsh** [comment]
For what it's worth, we do allow this:

```py
class Foo:
    """
    See:
        https://looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong.url.com
    """
```

That is: if the URL is on its own line, we don't flag it. Is that sufficient?

**Comment by jfcherng** [comment]
> For what it's worth, we do allow this:
> 
> ```python
> class Foo:
>     """
>     See:
>         https://looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong.url.com
>     """
> ```
> 
> That is: if the URL is on its own line, we don't flag it. Is that sufficient?

Probably OK for this case but for consistency, when it would be a little cumbersome if it's a short URL.

```py
class Foo:
    """
    See:
        https://short.url.com/
    """
```

v.s.

```py
class Foo:
    """
    @see https://short.url.com/
    """
```

**Comment by jat255** [comment]
Another issue where I've run into this when I recently started using ruff (wonderful tool, by the way!) is in RST text substitution links, something like the following:

```rst
    .. _named link on long line 1: https://qwertyyuiopasdfghjklzxcvbnm.qwertyyuiopasdfghjklzxcvbnm.org/
    .. _named link on long line 2: https://qwertyyuiopasdfghjklzxcvbnm.com/qwertyyuiopasdfghjklzxcvbnm.html#anchor_tag_1
```
